components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/DependencyNode.java

```diff
@@ -0,0 +1,172 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.uuf.api.config;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+/**
+ * A bean class that represents a node in the dependency tree.
+ *
+ * @since 1.0.0
+ */
+public class DependencyNode {
+
+    private String artifactId;
+    private String version;
+    private String contextPath;
+    private List<DependencyNode> dependencies;
+    private Set<String> allDependencies;
+
+    /**
+     * Creates a new dependency node.
+     */
+    public DependencyNode() {
+        this.dependencies = Collections.emptyList();
+    }
+
+    /**
+     * Returns the artifact ID of the UUF Component which is reflected by this node.
+     *
+     * @return artifact ID of the UUF Component
+     */
+    public String getArtifactId() {
+        return artifactId;
+    }
+
+    /**
+     * Sets the artifact ID of the UUF Component reflected by this node.
+     *
+     * @param artifactId artifact ID to be set
+     */
+    public void setArtifactId(String artifactId) {
+        this.artifactId = artifactId;
+    }
+
+    /**
+     * Returns the version of the UUF Component which is reflected by this node.
+     *
+     * @return version of the UUF Component
+     */
+    public String getVersion() {
+        return version;
+    }
+
+    /**
+     * Sets the version of the UUF Component reflected by this node.
+     *
+     * @param version version to be set
+     */
+    public void setVersion(String version) {
+        this.version = version;
+    }
+
+    /**
+     * Returns the context path of the UUF Component which is reflected by this node.
+     *
+     * @return context path of the UUF Component
+     */
+    public String getContextPath() {
+        return contextPath;
+    }
+
+    /**
+     * Sets the context path of the UUF Component reflected by this node. If the setting context path doesn't start with
+     * a '/', then it will be prepended.
+     *
+     * @param contextPath context path to be set
+     */
+    public void setContextPath(String contextPath) {
+        this.contextPath = contextPath.startsWith("/") ? contextPath : "/" + contextPath;
+    }
+
+    /**
+     * Returns the dependencies of the UUF Component which is reflected by this node.
+     *
+     * @return dependencies of the UUF Component
+     */
+    public List<DependencyNode> getDependencies() {
+        return dependencies;
+    }
+
+    /**
+     * Sets the dependencies list of the UUF Component reflected by this node.
+     *
+     * @param dependencies dependencies list to be set
+     */
+    public void setDependencies(List<DependencyNode> dependencies) {
+        if (dependencies == null) {
+            this.dependencies = Collections.emptyList();
+            this.allDependencies = Collections.emptySet();
```

Comment:
Move these two initialisations to class level or constructer and have the not null check and then the else logic.